exporter/datadogexporter/internal/translator/metrics_translator.go

```diff
@@ -243,10 +245,11 @@ func (t *Translator) getLegacyBuckets(
 		bucketTags = append(bucketTags, tags...)
 
 		count := float64(val)
+		startTs := uint64(p.StartTimestamp())
 		ts := uint64(p.Timestamp())
 		if delta {
 			consumer.ConsumeTimeSeries(ctx, fullName, Count, ts, count, bucketTags, host)
-		} else if dx, ok := t.prevPts.putAndGetDiff(fullName, bucketTags, ts, count); ok {
+		} else if dx, ok := t.prevPts.Diff(fullName, bucketTags, startTs, ts, count); ok {
```

Comment:
Same comment applies here (this case would be easier to debug but for consistency I would keep it as is)